### PR TITLE
chore(core): Add logging

### PR DESCRIPTION
**Pull request by renaynay**
Add more logs

### CODE REVIEW

**Comment by Wondertan** [comment]
We should wrap error instead and make the caller log the error out

File: core/exchange.go [hunk 1]
```diff
@@ -78,31 +79,36 @@ func (ce *Exchange) Get(ctx context.Context, hash libhead.Hash) (*header.Extende
 	log.Debugw("requesting header", "hash", hash.String())
 	block, err := ce.fetcher.GetBlockByHash(ctx, hash)
 	if err != nil {
+		log.Errorw("fetching block by hash", "hash", hash, "err", err)
 		return nil, err
 	}
 
 	comm, vals, err := ce.fetcher.GetBlockInfo(ctx, &block.Height)
 	if err != nil {
+		log.Errorw("fetching block info", "height", &block.Height, "err", err)
```
Comment (walldiss) [inline core/exchange.go:88]: Does the caller log errors? I wonder if we actually could wrap error here and log on caller side.

Comment (Wondertan) [reply core/exchange.go:88]: https://github.com/celestiaorg/celestia-node/pull/1961#issuecomment-1481863370

File: core/exchange.go [hunk 2]
```diff
@@ -116,17 +117,21 @@ func (ce *Exchange) Head(ctx context.Context) (*header.ExtendedHeader, error) {
...
 	// extend block data
 	eds, err := extendBlock(b.Data)
 	if err != nil {
+		log.Errorw("extending block data", "height", b.Header.Height, "err", err)
 		return nil, err
 	}
 	// create extended header
 	eh, err := ce.construct(ctx, &b.Header, &b.Commit, &b.ValidatorSet, eds)
 	if err != nil {
+		log.Errorw("constructing extended header", "height", b.Header.Height, "err", err)
```
Comment (Wondertan) [inline core/exchange.go:134]: wrap?

Comment (Wondertan) [reply core/exchange.go:134]: Every one in the method

**Review by vgonkivs — APPROVED** [review]

**Pull request merged**